From 9ccbe2351737e97ddd6ecc0c4744995b85664276 Mon Sep 17 00:00:00 2001
From: Veaceslav Falico <vfalico@redhat.com>
Date: Thu, 21 Aug 2014 11:28:26 -0400
Subject: [ethernet] netxen: lower NAPI weight

Message-id: <1408620515-703-2-git-send-email-vfalico@redhat.com>
Patchwork-id: 88511
O-Subject: [PATCH RHEL7 01/10] netxen: lower NAPI weight
Bugzilla: 1089356
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

RHBZ#1089356

Testing:
Locally and by QLogic (which supplied the patch)

Description:
Backport of the following commit:

commit 1e4a5282b4791fb9ba68478ecddae9642f9cfefb
Author: Michal Schmidt <mschmidt@redhat.com>
Date:   Fri Aug 23 15:41:09 2013 +0200

    netxen: lower NAPI weight

    Since commit 82dc3c63 ("net: introduce NAPI_POLL_WEIGHT")
    netif_napi_add() produces an error message if a NAPI poll weight
    greater than 64 is requested.

    Use the standard NAPI weight.

    v2: proper reference to the related commit

    Signed-off-by: Michal Schmidt <mschmidt@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/netxen/netxen_nic.h b/drivers/net/ethernet/qlogic/netxen/netxen_nic.h
index f40fb63..e8eff3e 100644
--- a/drivers/net/ethernet/qlogic/netxen/netxen_nic.h
+++ b/drivers/net/ethernet/qlogic/netxen/netxen_nic.h
@@ -1171,7 +1171,6 @@ typedef struct {
 
 #define NETXEN_DB_MAPSIZE_BYTES     0x1000
 
-#define NETXEN_NETDEV_WEIGHT 128
 #define NETXEN_ADAPTER_UP_MAGIC 777
 #define NETXEN_NIC_PEG_TUNE 0
 
diff --git a/drivers/net/ethernet/qlogic/netxen/netxen_nic_main.c b/drivers/net/ethernet/qlogic/netxen/netxen_nic_main.c
index 1d3f23c..5a119a3 100644
--- a/drivers/net/ethernet/qlogic/netxen/netxen_nic_main.c
+++ b/drivers/net/ethernet/qlogic/netxen/netxen_nic_main.c
@@ -197,7 +197,7 @@ netxen_napi_add(struct netxen_adapter *adapter, struct net_device *netdev)
  for (ring = 0; ring < adapter->max_sds_rings; ring++) {
   sds_ring = &recv_ctx->sds_rings[ring];
   netif_napi_add(netdev, &sds_ring->napi,
-    netxen_nic_poll, NETXEN_NETDEV_WEIGHT);
+    netxen_nic_poll, NAPI_POLL_WEIGHT);
  }
 
  return 0;
-- 
1.7.1